### DISCUSSION TITLE
images fetching so much but it's as the same?

### DISCUSSION THREAD

**Discussion by synasapmob** [community/community#147004] [Programming Help] [answered]
## My Question
hi guys, I'm trying to optimization my application with reduce fetch assets, 
in this video it's indicating when injected **element** network will refetch again?

the status **206** or **304** is good i think so,
but assets can't rendering fast because status now is 'pending'

```jsx
[1,2,3].map(() => <video src="as_the_same_url_but_it_call_again" />)
```

## Demo by video

1. refetch again

https://github.com/user-attachments/assets/42d2d631-a351-4a2b-8d8c-eb5fd289cbe6

3. status pending

https://github.com/user-attachments/assets/6305da6e-a6c2-4f87-96bf-f678a6d95dcb

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by synasapmob** [comment]
nobody help me, I'm cyring

**Comment by Logophoman** [comment]
What are you trying to do exactly? What is your Framework you're doing this with? Do you have code or are you just doing this in the console? Please provide a comprehensive description of the problem.

**Reply by synasapmob** [reply]
I'm using ReactJS + Vite, let's me show you more detail in below:

https://github.com/user-attachments/assets/a0b6b586-74c6-436d-88f6-3fb7d62fc12b

**Comment by tzachi81** [comment]
## The short answer
- Basically, you will fetch the sources once in a network call.
- Furthermore, If you want to optimize the data that it will not call the fetch each page refresh you can use [memoization](https://www.freecodecamp.org/news/memoization-in-javascript-and-react/), like UseMemo react hook.

## Elaborated example with React.js:
Pay attention that a video element can render one src at a time. 

I suggest to try this approach:
1. Create the video element once outside the .map() function.
2. Use the video tag property "onEnded" to change the index of the next source item in your array of video sources.
3. Replace the source index when the previous video is done playing. 
4. The change of current Index state will cause to re-render the video with the new source set in "src=" property

### Example if you use React functional component:

`
//example of currentIndex state management with useState hook:
const [currentSourceIndex, setCurrentSourceIndex] = useState(0);
.
.
.

// the return function of the VidePlayer component will be something like:
return (
<video
src={mySourcesArray[currentSourceIndex].src}
onEnded={updateCurrentSourceIndex}`
...
/>

**Reply by synasapmob** [reply, marked as answer]
> ## The short answer
> * Basically, you will fetch the sources once in a network call.
> * Furthermore, If you want to optimize the data that it will not call the fetch each page refresh you can use [memoization](https://www.freecodecamp.org/news/memoization-in-javascript-and-react/), like UseMemo react hook.
> 
> ## Elaborated example with React.js:
> Pay attention that a video element can render one src at a time.
> 
> I suggest to try this approach:
> 
> 1. Create the video element once outside the .map() function.
> 2. Use the video tag property "onEnded" to change the index of the next source item in your array of video sources.
> 3. Replace the source index when the previous video is done playing.
> 4. The change of current Index state will cause to re-render the video with the new source set in "src=" property
> 
> ### Example if you use React functional component:
> ` //example of currentIndex state management with useState hook: const [currentSourceIndex, setCurrentSourceIndex] = useState(0); . . .
> 
> // the return function of the VidePlayer component will be something like: return ( <video src={mySourcesArray[currentSourceIndex].src} onEnded={updateCurrentSourceIndex}` ... />

thanks sir, let me check again
